In [1]:
mkdir data

mkdir: cannot create directory ‘data’: File exists


In [1]:
from geoband.API import *
GetCompasData('SBJ_2107_004', '1', '1.5개시도_경찰서_관할경계.geojson')
GetCompasData('SBJ_2107_004', '2', '2.5개시도_치안안전센터_관할경계.geojson')
GetCompasData('SBJ_2107_004', '3', '3.체감안전도_설문지.txt')
GetCompasData('SBJ_2107_004', '4', '4.체감안전도_통계(2017~2019).csv')
GetCompasData('SBJ_2107_004', '5', '5.112신고.csv')
GetCompasData('SBJ_2107_004', '6', '6.범죄발생원표.csv')
GetCompasData('SBJ_2107_004', '7', '7.범죄검거원표.csv')
GetCompasData('SBJ_2107_004', '8', '8.범죄피의자원표.csv')
GetCompasData('SBJ_2107_004', '9', '9.5개시도_보안등현황.csv')
GetCompasData('SBJ_2107_004', '10', '10.5개시도_CCTV설치현황.csv')
GetCompasData('SBJ_2107_004', '11', '11.5개시도_교통사고현황.csv')
GetCompasData('SBJ_2107_004', '12', '12.경찰서별_지구대별 인원현황.csv')
GetCompasData('SBJ_2107_004', '13', '13.5개시도_화재발생통계.csv')
GetCompasData('SBJ_2107_004', '14', '14.5개시도_성연령별_총인구분포.csv')
GetCompasData('SBJ_2107_004', '15', '15.5개시도_1인가구수.csv')
GetCompasData('SBJ_2107_004', '16', '16.5개시도_인구데이터(외국인).csv')
GetCompasData('SBJ_2107_004', '17', '17.5개시도_기초수급자현황.csv')
GetCompasData('SBJ_2107_004', '18', '18.5개시도_자살율.csv')
GetCompasData('SBJ_2107_004', '19', '19.5개시도_최종학력통계.csv')
GetCompasData('SBJ_2107_004', '20', '20.5개시도_공원현황.csv')
GetCompasData('SBJ_2107_004', '21', '21.5개시도_유흥업소_단란주점현황.csv')
GetCompasData('SBJ_2107_004', '22', '22.5개시도_비상벨현황.geojson')
GetCompasData('SBJ_2107_004', '23', '23.5개시도_종관관측기상(2017~2020).csv')
GetCompasData('SBJ_2107_004', '24', '24.5개시도_도로명주소(건물).geojson')
GetCompasData('SBJ_2107_004', '25', '25.결과제출양식.csv')
GetCompasData('SBJ_2107_004', '26', '26.코드정의서.xlsx')

In [4]:
import pandas as pd
pd.read_csv('./data/25.결과제출양식.csv')

순번        경찰서명  종합체감안전도_점수   \n(전반적안전도*0.3) + (분야별안전도*0.7)  \
0    1    서울중부경찰서                                          NaN   
1    2    서울종로경찰서                                          NaN   
2    3   서울남대문경찰서                                          NaN   
3    4   서울서대문경찰서                                          NaN   
4    5    서울혜화경찰서                                          NaN   
5    6    서울용산경찰서                                          NaN   
6    7    서울성북경찰서                                          NaN   
7    8   서울동대문경찰서                                          NaN   
8    9    서울마포경찰서                                          NaN   
9   10   서울영등포경찰서                                          NaN   
10  11    서울성동경찰서                                          NaN   
11  12    서울동작경찰서                                          NaN   
12  13    서울광진경찰서                                          NaN   
13  14    서울서부경찰서                                          NaN   
14  15    서울강북경찰서                                          NaN   
15  16    서울금천경찰서                                          NaN   
16  17    서울중랑경찰서                                          NaN   
17  18    서울강남경찰서                                          NaN   
18  19    서울관악경찰서                                          NaN   
19  20    서울강서경찰서                                          NaN   
20  21    서울강동경찰서                                          NaN   
21  22    서울종암경찰서                                          NaN   
22  23    서울구로경찰서                                          NaN   
23  24    서울서초경찰서                                          NaN   
24  25    서울양천경찰서                                          NaN   
25  26    서울송파경찰서                                          NaN   
26  27    서울노원경찰서                                          NaN   
27  28    서울방배경찰서                                          NaN   
28  29    서울은평경찰서                                          NaN   
29  30    서울도봉경찰서                                          NaN   
30  31    서울수서경찰서                                          NaN   
31  32      세종경찰서                                          NaN   
32  33    수원중부경찰서                                          NaN   
33  34    수원남부경찰서                                          NaN   
34  35    수원서부경찰서                                          NaN   
35  36    창원중부경찰서                                          NaN   
36  37    창원서부경찰서                                          NaN   
37  38    마산중부경찰서                                          NaN   
38  39    마산동부경찰서                                          NaN   
39  40      진해경찰서                                          NaN   
40  41      진주경찰서                                          NaN   

    분야별 안전도\n(범죄안전도*0.343) + \n(교통사고안전도*0.305)\n + (법질서준수도*0.352)  \
0                                                 NaN               
1                                                 NaN               
2                                                 NaN               
3                                                 NaN               
4                                                 NaN               
5                                                 NaN               
6                                                 NaN               
7                                                 NaN               
8                                                 NaN               
9                                                 NaN               
10                                                NaN               
11                                                NaN               
12                                                NaN               
13                                                NaN               
14                                                NaN               
15                                                NaN               
16                                                NaN     

In [3]:
import pandas as pd
pd.read_csv('./data/4.체감안전도_통계(2017~2019).csv')

large_category medium_category  tot_f_safety  dept_safety  crime_safety  \
0     2017 상반기(연령대)    10대(만14세 이상)          74.3         73.3          81.5   
1     2017 상반기(연령대)             20대          70.1         69.3          76.9   
2     2017 상반기(연령대)             30대          69.6         68.9          76.8   
3     2017 상반기(연령대)             40대          70.7         70.2          77.2   
4     2017 상반기(연령대)             50대          72.1         71.5          78.3   
..              ...             ...           ...          ...           ...   
285  2019 하반기(경찰서별)         경남창원서부서          74.0         73.2          79.3   
286  2019 하반기(경찰서별)         경남마산중부서          75.4         74.9          80.4   
287  2019 하반기(경찰서별)         경남마산동부서          71.9         71.4          75.8   
288  2019 하반기(경찰서별)           경남진주서          75.1         74.9          79.2   
289  2019 하반기(경찰서별)           경남진해서          72.5         72.1          77.9   

      rob_satety_q1  mur_safety_q2  ta_safety_q3  raw_odder_q4  overall_q5  
0              80.1           82.8          70.9          66.9        76.5  
1              75.7           78.2          66.1          64.3        71.7  
2              75.7           77.9          63.9          65.2        71.1  
3              76.1           78.3          65.5          67.0        71.9  
4              77.7           79.0          68.0          67.6        73.3  
..              ...            ...           ...           ...         ...  
285            79.1           79.5          68.5          71.2        76.0  
286            80.8           80.0          70.8          73.2        76.5  
287            75.0           76.5          69.4          68.9        73.1  
288            78.7           79.6          70.3          74.6        75.6  
289            76.6           79.2          67.0          70.9        73.5  

[290 rows x 10 columns]

In [2]:
!mv *'.csv' ./data
!mv *'.geojson' ./data
!mv *'.xlsx' ./data
!mv *'.txt' ./data

In [1]:
import geojson
import geopandas as gpd
emergency_bell = gpd.read_file('./data/22.5개시도_비상벨현황.geojson')
emergency_bell.head(3)

/opt/app-root/lib/python3.6/site-packages/geopandas/_compat.py:110: UserWarning: The Shapely GEOS version (3.8.0-CAPI-1.13.1 ) is incompatible with the GEOS version PyGEOS was compiled with (3.9.0-CAPI-1.16.2). Conversions between both will be slow.
  shapely_geos_version, geos_capi_version_string


bell_us bell_set_fac       bell_loc  bell_set_yr  \
0      기타          화장실  오가낭뜰 근린공원(간이)         2017   
1      기타          화장실        어진동음악분수         2017   
2      기타          화장실         제천(간이)         2017   

                          geometry  
0  MULTIPOINT (127.24880 36.51230)  
1  MULTIPOINT (127.25660 36.50190)  
2  MULTIPOINT (127.25660 36.50190)

In [2]:
emergency_bell.info()

<class 'geopandas.geodataframe.GeoDataFrame'>
RangeIndex: 18334 entries, 0 to 18333
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype   
---  ------        --------------  -----   
 0   bell_us       18334 non-null  object  
 1   bell_set_fac  18334 non-null  object  
 2   bell_loc      18333 non-null  object  
 3   bell_set_yr   18334 non-null  int64   
 4   geometry      18334 non-null  geometry
dtypes: geometry(1), int64(1), object(3)
memory usage: 716.3+ KB


In [3]:
import folium
style_if = lambda x : {'color' : '#A9E2F3'}
Map = folium.Map(location=(36,128), zoom_start=7)
folium.GeoJson(emergency_bell, style_function = style_if).add_to(Map)
Map.save('bell.html')

In [4]:
Map.

AttributeError: 'Map' object has no attribute 'load'

In [8]:
import geojson
import geopandas as gpd
police_border = gpd.read_file('./data/1.5개시도_경찰서_관할경계.geojson')
police_border.head(3)

NAME PNAME                                           geometry
0    세종경찰서   충남청  MULTIPOLYGON (((127.17202 36.73106, 127.17202 ...
1    진주경찰서   경남청  MULTIPOLYGON (((128.26697 35.12927, 128.26697 ...
2  창원서부경찰서   경남청  MULTIPOLYGON (((128.63363 35.22152, 128.63357 ...

In [ ]:
gpd.

In [9]:
import folium
style_if = lambda x : {'color' : '#A9E2F3'}
Map = folium.Map(location=(36,128), zoom_start=7)
folium.GeoJson(police_border).add_to(Map)
Map

In [3]:
gpd.read_file('./data/2.5개시도_치안안전센터_관할경계.geojson')

NAME    PSNAME                                           geometry
0     삼계파출소   마산동부경찰서  MULTIPOLYGON (((128.49829 35.27500, 128.49876 ...
1     중리파출소   마산동부경찰서  MULTIPOLYGON (((128.51822 35.24046, 128.51824 ...
2     교방파출소   마산중부경찰서  MULTIPOLYGON (((128.57035 35.21196, 128.57039 ...
3     구산파출소   마산중부경찰서  MULTIPOLYGON (((128.53721 35.13687, 128.53732 ...
4     남성파출소   마산중부경찰서  MULTIPOLYGON (((128.57486 35.20959, 128.57497 ...
..      ...       ...                                                ...
313   상일파출소   서울강동경찰서  MULTIPOLYGON (((127.15610 37.54608, 127.15617 ...
314  용산역파출소   서울용산경찰서  MULTIPOLYGON (((126.98037 37.53793, 126.98044 ...
315   홍익지구대   서울마포경찰서  MULTIPOLYGON (((126.91769 37.55853, 126.91854 ...
316   당산지구대  서울영등포경찰서  MULTIPOLYGON (((126.90299 37.54263, 126.90280 ...
317   거여파출소   서울송파경찰서  MULTIPOLYGON (((127.15254 37.49293, 127.15244 ...

[318 rows x 3 columns]

In [4]:
import pandas as pd

In [5]:
security_total =  pd.read_csv('./data/4.체감안전도_통계(2017~2019).csv')
security_total.head(3)

large_category medium_category  tot_f_safety  dept_safety  crime_safety  \
0  2017 상반기(연령대)    10대(만14세 이상)          74.3         73.3          81.5   
1  2017 상반기(연령대)             20대          70.1         69.3          76.9   
2  2017 상반기(연령대)             30대          69.6         68.9          76.8   

    rob_satety_q1  mur_safety_q2  ta_safety_q3  raw_odder_q4  overall_q5  
0            80.1           82.8          70.9          66.9        76.5  
1            75.7           78.2          66.1          64.3        71.7  
2            75.7           77.9          63.9          65.2        71.1

※수식
<p>종합체감안전도점수 = (전반적안전도*0.3) + (분야별안전도*0.7)
<p>분야별 안전도 = (범죄안전도*0.343) + (교통사고안전도*0.305) + (법질서준수도*0.352)
<p>범죄 안전도 = (절도폭력안전도+강도살인안전도)/2
<p>절도폭력 안전도_문항1 = (문항1 점수 * 10)
<p>강도살인 안전도_문항2 = (문항2 점수 * 10)
<p>교통사고 안전도_문항3 = (문항3 점수*10)
<p>법질서 준수도_문항4 = (문항4 점수 * 10)
<p>전반적 안전도_문항5 = (문항5 점수 * 10)

In [63]:
crime_rise =  pd.read_csv('./data/6.범죄발생원표.csv')
crime_rise.head(3)

/opt/app-root/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3072: DtypeWarning: Columns (4,6,8,10) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


jur_stn        crm crm_wthr crm_clue vic_sx vic_age rm_loc       crm_tm  \
0   서울수서경찰서   위조외국통화행사       미상       진정     불상     NaN     은행  09:00-11:59   
1  서울영등포경찰서    도로교통법위반       맑음     타인신고     불상     NaN     노상  21:00-23:59   
2   서울양천경찰서  209015100       미상    피해자신고     남자   60세초과     노상           미상   

   crm_date 5m_crm_yn crm_loc  
0  20170101               NaN  
1  20170101               NaN  
2  20170101               NaN

In [64]:
crime_rise.tail(3)

jur_stn   crm crm_wthr crm_clue vic_sx vic_age rm_loc       crm_tm  \
1073219  경남마산동부경찰서  재물손괴       미상    피해자신고    NaN   60세이하    NaN  12:00∼14:59   
1073220  경남마산중부경찰서    사기       미상       진정    NaN   50세이하    NaN  12:00∼14:59   
1073221  경남마산동부경찰서  강제추행       미상       고소    NaN   30세이하    NaN  21:00∼23:59   

          crm_date  5m_crm_yn crm_loc  
1073219  9/17/2019         폭력      주택  
1073220  9/14/2019                 기타  
1073221  5/16/2019  강간 및 강제추행      기타

In [75]:
len(crime_rise['jur_stn'].value_counts())

41

In [73]:
crime_rise['jur_stn'].value_counts()

경기수원남부경찰서    53750
서울송파경찰서      49093
서울강남경찰서      46842
서울영등포경찰서     46399
서울강서경찰서      40468
서울관악경찰서      40070
서울마포경찰서      39138
경기수원중부경찰서    37622
서울서초경찰서      36992
서울구로경찰서      34157
서울노원경찰서      30897
서울중랑경찰서      30726
서울광진경찰서      29935
경기수원서부경찰서    29780
서울동대문경찰서     29263
서울강동경찰서      28560
서울수서경찰서      27889
서울용산경찰서      27238
서울양천경찰서      26740
서울금천경찰서      26200
서울강북경찰서      26052
경남진주경찰서      25432
서울동작경찰서      24326
서울서대문경찰서     23697
서울성동경찰서      22481
경남창원중부경찰서    21041
서울중부경찰서      17793
경남마산동부경찰서    17756
서울은평경찰서      17493
서울도봉경찰서      17278
경남창원서부경찰서    16751
서울종로경찰서      15650
충남세종경찰서      14186
서울혜화경찰서      13949
서울남대문경찰서     13699
서울성북경찰서      13641
경남마산중부경찰서    13593
경남진해경찰서      13471
서울종암경찰서      12102
서울방배경찰서      10888
서울서부경찰서      10184
Name: jur_stn, dtype: int64

In [7]:
len(crime_rise)

1073222

In [76]:
from datetime import datetime
crime_rise['crm_date'] = crime_rise['crm_date'].apply(lambda x : datetime.strptime(str(x), '%Y%m%d').strftime('%m/%d/%Y') if len(str(x)) == 8 and type(x) == int else x)

In [77]:
crime_rise['crm_date'] = crime_rise['crm_date'].apply(lambda x : datetime.strptime(str(x), '%Y%m%d').strftime('%m/%d/%Y') if str(x).count('/') == 0 else x)

In [78]:
crime_rise['crm_date'] = crime_rise['crm_date'].apply(lambda x : datetime.strptime(str(x), '%m/%d/%Y').strftime('%m/%d/%Y'))

In [79]:
crime_rise['crm_date'].value_counts()

01/01/2017    2641
07/01/2017    1772
06/01/2017    1757
09/01/2017    1717
08/01/2017    1687
              ... 
01/10/2014       1
07/28/2015       1
02/08/2016       1
06/26/2014       1
02/18/2014       1
Name: crm_date, Length: 2378, dtype: int64

In [80]:
crime_rise['crm_date'].sort_values(ascending=False)

829839     12/31/2019
590968     12/31/2018
590956     12/31/2018
590957     12/31/2018
590958     12/31/2018
              ...    
1061358    01/01/2008
1044304    01/01/2006
1031580    01/01/2003
1052546    01/01/2002
1050317    01/01/1992
Name: crm_date, Length: 1073222, dtype: object

In [82]:
len(crime_rise[crime_rise['crm_date'].str.contains(str(2017))] )

393879

In [83]:
len(crime_rise[crime_rise['crm_date'].str.contains(str(2018))] )

367619

In [84]:
len(crime_rise[crime_rise['crm_date'].str.contains(str(2019))] )

306742

In [85]:
len(crime_rise[crime_rise['crm_date'].str.contains(str(2020))] )

0

In [157]:
crime_rise['crm_date'].sort_values()

1050317    01/01/1992
1052546    01/01/2002
1031580    01/01/2003
1044304    01/01/2006
1061358    01/01/2008
              ...    
590958     12/31/2018
590957     12/31/2018
590956     12/31/2018
590968     12/31/2018
829839     12/31/2019
Name: crm_date, Length: 1073222, dtype: object

In [127]:
crime_2017 = crime_rise[crime_rise['crm_date'].str.contains(str(2017))]
crime_2018 = crime_rise[crime_rise['crm_date'].str.contains(str(2018))] 
crime_2019 = crime_rise[crime_rise['crm_date'].str.contains(str(2019))] 

# 범죄 발생 기록 2017~2019도만 남김
crime_rise_df = pd.concat([crime_2017, crime_2018, crime_2019])
crime_rise_df.head(3)

jur_stn        crm crm_wthr crm_clue vic_sx vic_age rm_loc       crm_tm  \
0   서울수서경찰서   위조외국통화행사       미상       진정     불상     NaN     은행  09:00-11:59   
1  서울영등포경찰서    도로교통법위반       맑음     타인신고     불상     NaN     노상  21:00-23:59   
2   서울양천경찰서  209015100       미상    피해자신고     남자   60세초과     노상           미상   

     crm_date 5m_crm_yn crm_loc  
0  01/01/2017               NaN  
1  01/01/2017               NaN  
2  01/01/2017               NaN

In [156]:
crime_rise_df['crm_date'].sort_values()

0          01/01/2017
1758       01/01/2017
1759       01/01/2017
1760       01/01/2017
1761       01/01/2017
              ...    
590715     12/31/2018
590716     12/31/2018
1017843    12/31/2018
929890     12/31/2018
829839     12/31/2019
Name: crm_date, Length: 1068240, dtype: object

In [119]:
# 5대 범죄 해당 여부  : 폭력 / 절도 / 강간 및 강제추행 / 강도 / 살인
crime_rise_df['5m_crm_yn'].value_counts()

             685688
폭력           209971
절도           150045
강간 및 강제추행     21516
강도              519
살인              501
Name: 5m_crm_yn, dtype: int64

In [120]:
crime_rise_df.groupby(['jur_stn','5m_crm_yn']).count()

crm  crm_wthr  crm_clue  vic_sx  vic_age  rm_loc  \
jur_stn   5m_crm_yn                                                       
경기수원남부경찰서            36195     36195     36195   36195    20730   36195   
          강간 및 강제추행    789       789       789     789      769     789   
          강도            22        22        22      22       22      22   
          살인            21        21        21      21       20      21   
          절도          6856      6856      6856    6856     6769    6856   
...                    ...       ...       ...     ...      ...     ...   
충남세종경찰서   강간 및 강제추행    193       193       193     193      191     193   
          강도             7         7         7       7        7       7   
          살인            12        12        12      12       11      12   
          절도          1911      1911      1911    1911     1877    1911   
          폭력          2196      2196      2196    2196     2015    2196   

                     crm_tm  crm_date  crm_loc  
jur_stn   5m_crm_yn                             
경기수원남부경찰서             36195     36195        0  
          강간 및 강제추행     789       789        0  
          강도             22        22        0  
          살인             21        21        0  
          절도           6856      6856        0  
...                     ...       ...      ...  
충남세종경찰서   강간 및 강제추행     193       193        0  
          강도              7         7        0  
          살인             12        12        0  
          절도           1911      1911        0  
          폭력           2196      2196        0  

[246 rows x 9 columns]

In [155]:
crime_rise_df.groupby(['crm_date','jur_stn']).count()

crm  crm_wthr  crm_clue  vic_sx  vic_age  rm_loc  \
crm_date   jur_stn                                                       
01/01/2017 경기수원남부경찰서   92        92        92      92       67      92   
           경기수원서부경찰서   46        46        46      46       40      46   
           경기수원중부경찰서   68        68        68      68       50      68   
           경남마산동부경찰서   33        33        33      24       20       0   
           경남마산중부경찰서   19        19        19      16       15       0   
...                   ...       ...       ...     ...      ...     ...   
12/31/2018 서울중랑경찰서     18        18        18       0       14      18   
           서울중부경찰서     13        13        13       0       13      13   
           서울혜화경찰서      5         5         5       0        2       5   
           충남세종경찰서     10        10        10      10        7      10   
12/31/2019 서울구로경찰서      1         1         1       1        1       1   

                      crm_tm  5m_crm_yn  crm_loc  date  
crm_date   jur_stn                                      
01/01/2017 경기수원남부경찰서      92         92        0    92  
           경기수원서부경찰서      46         46        0    46  
           경기수원중부경찰서      68         68        0    68  
           경남마산동부경찰서      33         33       33    33  
           경남마산중부경찰서      19         19       19    19  
...                      ...        ...      ...   ...  
12/31/2018 서울중랑경찰서        18         18        0    18  
           서울중부경찰서        13         13        0    13  
           서울혜화경찰서         5          5        0     5  
           충남세종경찰서        10         10        0    10  
12/31/2019 서울구로경찰서         1          1        0     1  

[44692 rows x 10 columns]

In [113]:
crime_2017['crm'].value_counts().head(20)

사기                             51497
절도                             46746
폭행                             41746
209015100                      33914
도로교통법위반(음주운전)                  31838
재물손괴                           13741
점유이탈물횡령                        13008
도로교통법위반(무면허운전)                 10315
상해                              6820
업무방해                            6422
모욕                              5828
자동차관리법위반                        4757
저작권법위반                          4690
교통사고처리특례법위반                     4241
강제추행                            3960
도로교통법위반(사고후미조치)                 3833
교통사고처리특례법위반(치상)                 3692
도로교통법위반                         3570
자동차손해배상보장법위반                    3214
성폭력범죄의처벌등에관한특례법위반(카메라등이용촬영)     2694
Name: crm, dtype: int64

In [29]:
crime_2018['crm'].value_counts()

사기                 54972
절도                 45551
교통사고처리특례법위반(치상)    41100
폭행                 39122
도로교통법위반(음주운전)      23665
                   ...  
면허증불실기재                1
강도살인미수                 1
컴퓨터등장애업무방해방조           1
간음약취미수                 1
235040400              1
Name: crm, Length: 947, dtype: int64

In [30]:
crime_2019['crm'].value_counts()

절도                                41853
사기                                40668
교통사고처리특례법위반(치상)                   39157
폭행                                34396
도로교통법위반(음주운전)                     17596
                                  ...  
부당이득                                  1
정보통신망이용촉진및정보보호등에관한법률위반(명예훼손)교사        1
공익법인의설립운영에관한법률위반                      1
음반·비디오물및게임물에관한법률위반                    1
특수강요                                  1
Name: crm, Length: 847, dtype: int64

In [31]:
crime_2020['crm'].value_counts()

Series([], Name: crm, dtype: int64)

In [9]:
crime_rise['crm'].value_counts()

사기                 149049
절도                 134420
폭행                 115429
교통사고처리특례법위반(치상)     83953
도로교통법위반(음주운전)       73185
                    ...  
경매방해방조                  1
모해증거변조                  1
업무상승낙낙태치사               1
자동차불법사용미수               1
자격모용유가증권작성              1
Name: crm, Length: 1244, dtype: int64

In [87]:
report_112 = pd.read_csv('./data/5.112신고.csv')
report_112.head(3)

date jur_stn report_sx inc_info
0  20180603.0    서울서부        남성     분실습득
1  20180603.0    서울송파        여성       폭력
2  20180603.0    서울강서        남성       소음

In [89]:
report_112['date'] = report_112['date'].apply(lambda x : datetime.strptime(str(int(x)), '%Y%m%d').strftime('%m/%d/%Y'))
report_112['date']

0          06/03/2018
1          06/03/2018
2          06/03/2018
3          06/03/2018
4          06/03/2018
              ...    
9229106    05/31/2021
9229107    05/31/2021
9229108    05/31/2021
9229109    05/31/2021
9229110    05/31/2021
Name: date, Length: 9229111, dtype: object

In [86]:
report_112['inc_info'].value_counts()

기타형사범         1296049
상담문의           889824
보호조치           801201
시비             646502
교통사고           583157
내용확인불가         533468
위험방지           482305
폭력             395801
소음             377366
행패소란           339712
교통불편           298216
분실습득           233298
기타_타기관         217360
절도             200585
실종(실종아동 등)     191318
주취자            189416
교통위반           171336
가정폭력           150383
비상벨            137012
기타경범           123585
청소년비행           91575
무전취식승차          81767
도박              71238
구조요청            64881
자살              62558
재물손괴            57684
풍속영업            53862
서비스요청           51706
화재              47002
사기              37248
데이트폭력           36079
가출 등            28953
성폭력             27499
수배불심자           25533
변사자             22106
주거침입            13304
아동학대(가정내)        7075
FTX              6957
협박               6939
경비업체요청           6743
스토킹              5530
위험동물             4810
노점상              3774
학교폭력             3635
납치감금             2655
아동학대(기타)  

In [90]:
# 2018.06 ~ 2020
report_2018 = report_112[report_112['date'].str.contains(str(2018))]
report_2019 = report_112[report_112['date'].str.contains(str(2019))]
report_2020 = report_112[report_112['date'].str.contains(str(2020))]

In [92]:
len(report_2018)

1845592

In [93]:
len(report_2019)

3243193

In [94]:
len(report_2020)

2991829

In [95]:
# 2017.01 ~ 2019.12
crime_arrest = pd.read_csv('./data/7.범죄검거원표.csv')
crime_arrest.head(3)

date    jur_stn            crm
0  201701  경남마산중부경찰서             사기
1  201701  경남마산중부경찰서       경범죄처벌법위반
2  201612  경남마산중부경찰서  도로교통법위반(음주운전)

In [108]:
crime_arrest['date'] = crime_arrest['date'].apply(lambda x : str(x))
crime_arrest_2016 = crime_arrest[crime_arrest['date'].str.contains('2016')].index
# 2016 drop
crime_arrest.drop(crime_arrest_2016, inplace=True)
len(crime_arrest)

928310

In [109]:
crime_arrest['crm'].value_counts().head(20)

사기                 136613
폭행                 110737
교통사고처리특례법위반(치상)     81880
도로교통법위반(음주운전)       75697
절도                  68627
209015100           35053
재물손괴                22018
도로교통법위반(무면허운전)      21062
업무방해                18340
상해                  17151
자동차관리법위반            14356
모욕                  13250
점유이탈물횡령             11178
강제추행                10496
도로교통법위반              8671
공무집행방해               7398
교통사고처리특례법위반          7191
병역법위반                6663
특수절도                 6518
횡령                   6457
Name: crm, dtype: int64

In [ ]:
# 검거 / 발생  ==> 검거율?

In [153]:
crime_rise_df['date'] = crime_rise_df['crm_date'].apply(lambda x : str(x.split('/')[2]) + str(x.split('/')[0]))
crime_rise_df['date'].sort_values()

0         201701
20330     201701
20329     201701
20328     201701
20327     201701
           ...  
825964    201912
825963    201912
825962    201912
825973    201912
829838    201912
Name: date, Length: 1068240, dtype: object

In [133]:
crime_arrest_cnt = crime_arrest.groupby(['date','jur_stn']).count()
crime_arrest_cnt

crm
date   jur_stn        
201701 경기수원남부경찰서  1119
       경기수원서부경찰서   780
       경기수원중부경찰서   944
       경남마산동부경찰서   489
       경남마산중부경찰서   280
...                ...
201912 서울종암경찰서     337
       서울중랑경찰서     675
       서울중부경찰서     337
       서울혜화경찰서     257
       충남세종경찰서     396

[1476 rows x 1 columns]

In [132]:
crime_rise_cnt = pd.DataFrame(crime_rise_df.groupby(['date', 'jur_stn']).count()['crm'])
crime_rise_cnt

crm
date   jur_stn        
201701 경기수원남부경찰서  1488
       경기수원서부경찰서   981
       경기수원중부경찰서  1155
       경남마산동부경찰서   523
       경남마산중부경찰서   334
...                ...
201912 서울종암경찰서     108
       서울중랑경찰서     216
       서울중부경찰서      95
       서울혜화경찰서      82
       충남세종경찰서      72

[1476 rows x 1 columns]

In [144]:
# 검거율 ?
crime_arrest_rate = crime_arrest_cnt / crime_rise_cnt
crime_arrest_rate.head(3)

crm
date   jur_stn            
201701 경기수원남부경찰서  0.752016
       경기수원서부경찰서  0.795107
       경기수원중부경찰서  0.817316

In [145]:
crime_arrest_rate.tail(3)

crm
date   jur_stn          
201912 서울중부경찰서  3.547368
       서울혜화경찰서  3.134146
       충남세종경찰서  5.500000

In [146]:
crime_arrest_rate = crime_arrest_rate.reset_index()
crime_arrest_rate.head(3)

date    jur_stn       crm
0  201701  경기수원남부경찰서  0.752016
1  201701  경기수원서부경찰서  0.795107
2  201701  경기수원중부경찰서  0.817316

In [158]:
# 2017~ 2019
crime_suspect = pd.read_csv('./data/8.범죄피의자원표.csv')
crime_suspect.head(3)

jur_stn    date      crm  ofn_age ofn_sx ofn_edu ofn_sol
0  서울서초경찰서  201709    의료법위반       58      여     NaN     NaN
1  서울은평경찰서  201710  대외무역법위반        0     불상     NaN     NaN
2  서울은평경찰서  201710  대외무역법위반       52      여    대학졸업      상류

In [160]:
crime_suspect.groupby(['date','jur_stn']).count()

crm  ofn_age  ofn_sx  ofn_edu  ofn_sol
date   jur_stn                                          
201403 경남창원중부경찰서    3        3       3        0        0
201604 경기수원남부경찰서    1        1       1        1        1
       서울구로경찰서      3        3       3        3        3
201605 서울관악경찰서      1        1       1        1        1
       서울구로경찰서      1        1       1        1        1
...               ...      ...     ...      ...      ...
201912 서울종암경찰서    401      401     401      246      246
       서울중랑경찰서    991      991     991      596      596
       서울중부경찰서    508      508     508      256      256
       서울혜화경찰서    307      307     307      197      197
       충남세종경찰서    582      582     582      298      299

[1532 rows x 5 columns]

In [163]:
# 17년 1월부터
crime_suspect = crime_suspect[crime_suspect['date'] >= 201701]
crime_suspect.sort_values('date')

jur_stn    date                     crm  ofn_age ofn_sx ofn_edu  \
156569     서울강남경찰서  201701             교통사고처리특례법위반       75      남   중학교졸업   
123570     서울서부경찰서  201701                 아동복지법위반       54      남    대학졸업   
123571     서울서부경찰서  201701                    재물손괴       66      남     NaN   
123664     서울서부경찰서  201701                      사기        0     불상     NaN   
123665     서울서부경찰서  201701                    강제추행       38      남     NaN   
...            ...     ...                     ...      ...    ...     ...   
965879     서울도봉경찰서  201912                      사기       50      남  고등학교졸업   
965878     서울도봉경찰서  201912                      사기       53      남     NaN   
965877     서울도봉경찰서  201912          도로교통법위반(무면허운전)       46      남   대학원졸업   
1152573  경기수원중부경찰서  201912  특정범죄가중처벌등에관한법률위반(도주치상)       29      남    대학중퇴   
809675     서울금천경찰서  201912                    재물손괴       62      남    대학졸업   

        ofn_sol  
156569       하류  
123570       중류  
123571      NaN  
123664      NaN  
123665      NaN  
...         ...  
965879       중류  
965878      NaN  
965877       하류  
1152573      하류  
809675       하류  

[1289979 rows x 7 columns]

In [168]:
crime_suspect['date'] = crime_suspect['date'].apply(lambda x :str(x))

/opt/app-root/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [169]:
crime_suspect_cnt = pd.DataFrame(crime_suspect.groupby(['date','jur_stn']).count()['crm'])
crime_suspect_cnt

crm
date   jur_stn        
201701 경기수원남부경찰서  1631
       경기수원서부경찰서  1002
       경기수원중부경찰서  1278
       경남마산동부경찰서   609
       경남마산중부경찰서   324
...                ...
201912 서울종암경찰서     401
       서울중랑경찰서     991
       서울중부경찰서     508
       서울혜화경찰서     307
       충남세종경찰서     582

[1476 rows x 1 columns]

In [171]:
# 검거율 ?
crime_arrest_rate = crime_arrest_cnt / crime_suspect_cnt
crime_arrest_rate

crm
date   jur_stn            
201701 경기수원남부경찰서  0.686082
       경기수원서부경찰서  0.778443
       경기수원중부경찰서  0.738654
       경남마산동부경찰서  0.802956
       경남마산중부경찰서  0.864198
...                    ...
201912 서울종암경찰서    0.840399
       서울중랑경찰서    0.681130
       서울중부경찰서    0.663386
       서울혜화경찰서    0.837134
       충남세종경찰서    0.680412

[1476 rows x 1 columns]

In [172]:
print(len(crime_suspect))
print(len(crime_rise_df))

1289979
1068240
